lib/rubygems/executor.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require 'open-uri'
+require 'json'
+require 'rubygems/platform'
+
+module Gem
+  module Web
+    class Executor
+
+      def supported_options_from_metadata
+        ["source_code_uri", "documentation_uri"]
+      end
+
+      def open_page(gem, options)
+        spec = Gem::Specification.find_by_name(gem)
+
+        if options[:sourcecode]
+          get_info_from_metadata(spec, "source_code_uri")
+        elsif options[:doc]
+          get_info_from_metadata(spec, "documentation_uri")
+        elsif options[:webpage]
```

Comment:
That's a good point. I only left it 'cuz you guys could want to change the default option. Since we're agreeing with homepage being the default, I'll remove this `elsif`